lib/runtime/src/distributed.rs

```diff
@@ -252,6 +253,29 @@ impl DistributedRuntime {
                         .system_status_server
                         .set(Arc::new(system_status_server_info))
                         .expect("System status server info should only be set once");
+
+                    // Register metrics endpoint with discovery
+                    // Use "system" namespace for the system status server's metrics endpoint
+                    let advertise_host = system_status_server::resolve_advertise_host(&host);
+                    let metrics_url = format!("http://{}:{}/metrics", advertise_host, addr.port());
+                    let metrics_spec = crate::discovery::DiscoverySpec::MetricsEndpoint {
+                        namespace: "system".to_string(),
```

Comment:
question: could we use the same namespace as the other endpoints?

Reply:
AFAIK we don't have this until someone calls `drt.namespace(...)` from the user side. The system server is created when we construct `DistributedRuntime(...)`. We could pass it in but that seems messy...